### DISCUSSION TITLE
How to set an environment variable for just the following command?

### DISCUSSION THREAD

**Discussion by halloleo** [xonsh/xonsh#4979] [Q&A] [answered]
This might be asked before, but I could not find it in the discussions or via a Google search:

I want to call `history show all` with another value for $XONSH_DATA_DIR.  So _just for the `history` command I want to change $XONSH_DATA_DIR. I tried bash-like:

    $XONSH_DATA_DIR = '/new/path' history show all

but this didn't work for me.

How can I set an environment variable for just the following command?

**Comment by halloleo** [comment, marked as answer]
Ups, found the answer myself: The bash like approach _does_ work -- when I make sure there are _no spaces in the definition of the environment variable_ in front of the command:

    $XONSH_DATA_DIR='/new/path' history show all

Sorry to have bothered the community. Happy xonshing!

**Reply by anki-code** [reply]
The alternative way for scripting from [xonsh-cheatsheet](https://github.com/anki-code/xonsh-cheatsheet):
```python
$VAR = 'value'
print($VAR)
with ${...}.swap(VAR='another value', NEW_VAR='new value'):  # Change VAR for commands block (xonsh way)
# with __xonsh__.env.swap(VAR='another value', NEW_VAR='new value'):  # Python way
    print($VAR)
print($VAR)
#value
#another value
#value
```
